src/downloader/YoutubeDownloader.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+from src.downloader.iDownloader import iDownloader, DownloadException
+from src.data.VideoItem import VideoItem
+import youtube_dl
+
+
+class YoutubeDownloader(iDownloader):
+
+    def __init__(self):
+        self.ydl = youtube_dl.YoutubeDL({
+            'nocheckcertificate': True,
+            'progress_hooks': [self.on_download_callback]
+        })
+
+    async def download(self, link:str) -> VideoItem:
+        self.ydl.download([link])
+        if self.file_name is None:
+            raise DownloadException()
```

Comment:
I think it'd be good to also output which link failed to download, either with a print statement or passing it into the DownloadException